.github/actions/emit-metrics/action.yml

```diff
@@ -0,0 +1,74 @@
+name: Emit Metrics
+description: Emit metrics from another workflow/action using the kat tool
+inputs:
+  namespace:
+    description: The CloudWatch namespace in which to emit metrics
+    required: true
+  dimensions:
+    description: |-
+      The dimensions to include with emitted metrics, as a collection of `name=value` pairs. Multiple dimensions are
+      delimited by newlines (`\n`) and whitespace is trimmed from before/after each `name=value` pair. When passing
+      multiple dimensions, make sure to use YAML's block literal syntax (`|` or `|-`).
+
+      For example:
+      ```
+      dimensions: |
+        Artifact=foo-artifact
+        Platform=Kotlin/JVM
+        Variant=External
+      ```
+    required: true
+  metrics:
+    description: |-
+      The metrics to emit into the given namespace and using the specified dimensions. Individual metrics are written in
+      the form of `<name>:<value>[:<unit>]` where:
+      * <name> is the metric name and may include spaces but not colons (`:`)
+      * <value> is the numeric value of the metric
+      * <unit> is the CloudWatch unit name (if omitted, `None` is assumed)
+
+      Multiple metrics are delimited by newlines (`\n`) and whitespace is trimmed from before/after each metric
+      specification. When passing multiple metrics, make sure to use YAML's block literal syntax (`|` or `|-`).
+
+      For example:
+      ```
+      metrics: |
+        BuildTime:4.532:Seconds
+        BuildSucceeded:1:Count
+      ```
+    required: true
+
+runs:
+  using: composite
+  steps:
+    - name: Emit Metrics
+      shell: bash
+      run: |
```

Comment:
This task should first check if the kat tool has been properly set up, and return an error if it's not available

Reply:
Is there a way we can idempotently invoke the `setup-kat` action? Then we can invoke it everywhere we use kat to ensure it's set up correctly

Reply:
I looked but didn't find an way to run an action (e.g., **setup-kat**) _only_ when it hasn't already been run. I suppose we could modify **setup-kat** to be idempotent by seeing if **kat** is present already in the path. But **setup-kat** itself requires **aws-actions/configure-aws-credentials** to have been run as well, for which we cannot so easily control idempotency.

Let me play with some options and see if we can do better.

Reply:
OK I was able to add **kat** verification to the action. I could probably cheese credentials verifications too but I've filed [a feature request to add idempotency to **configure-aws-credentials**](https://github.com/aws-actions/configure-aws-credentials/issues/1288) and I think that'll be the better approach long-term.